### PR TITLE
ctags precheck

### PR DESCRIPTION
**Pull request by Adman**
- check if ctags command exists
- check if ctags generated output compatible with ambu

Signed-off-by: Adrian Matejov a.matejov@centrum.sk

### CODE REVIEW

File: ambu [hunk 1]
```diff
@@ -21,8 +21,18 @@ if [ ! -d "$CURRENTDIR" ]; then
     exit 1
 fi
 
-mkdir -p $BUILDDIR
+command -v ctags >/dev/null 2>&1 || {
+        echo >&2 "ctags required for building your project. Please install it!"
```
Comment (mrshu) [inline ambu]: @Adman it seems that you are missing "is" here.

File: ambu [hunk 2]
```diff
@@ -21,8 +21,18 @@ if [ ! -d "$CURRENTDIR" ]; then
...
+        exit 1
+}
+
 ctags --language-force=c --exclude=$PROJECTNAME *.ino
+if ! grep -q "$(printf '_TAG_PROGRAM_NAME\tExuberant Ctags\t//')" tags; then
+    echo "ctags generated output not compatible with ambu. Please reinstall it!"
```
Comment (mrshu) [inline ambu]: @Adman I am not convinced reinstalling ctags would help here. Would it make more sense to provide information on how to install `ctags` that ambu expects?

**Comment by mrshu** [comment]
@Adman Just two small nitpicks, otherwise this looks good to me!

File: ambu (hunk 2 above)
Comment (Adman) [inline ambu]: @mrshu the question is that how it expects it to be installed? From repositories? From elsewhere?

Comment (mrshu) [inline ambu]: @Adman This is the correct question to ask I believe. I believe that your comment here should note that it is normally called `excuberant-ctags` in repositories of most distributions, and that there is also a website whose link you might want to attach to this comment too.

Comment (mrshu) [inline ambu]: @Adman One more thing, if the comment above is being forwarded to stderr, should this be forwarded there too?

**Comment by Adman** [comment]
@mrshu fixed

File: ambu [hunk 3]
```diff
@@ -21,8 +21,19 @@ if [ ! -d "$CURRENTDIR" ]; then
     exit 1
 fi
 
-mkdir -p $BUILDDIR
+command -v ctags >/dev/null 2>&1 || {
+        echo >&2 "ctags is required for building your project. Please install it!"
```
Comment (mrshu) [inline ambu]: @Adman any reason for a different amount of whitespace than on lines 31 to 33?

**Comment by mrshu** [comment]
@Adman This is really really close, sorry for all the nagging.

**Comment by Adman** [comment]
@mrshu there you go

**Comment by mrshu** [comment]
@Adman awesome, thanks!

**Pull request merged**